src/lens.jl

```diff
@@ -47,16 +47,20 @@ Concrete subtypes of `Lens` have to implement
 These must be pure functions, that satisfy the three lens laws:
 
 ```jldoctest; output = false, setup = :(using Setfield; obj = (a="A", b="B"); lens = @lens _.a; val = 2; val1 = 10; val2 = 20)
-@assert get(set(obj, lens, val), lens) == val
+@assert get(set(obj, lens, val), lens) ≅ val
         # You get what you set.
-@assert set(obj, lens, get(obj, lens)) == obj
+@assert set(obj, lens, get(obj, lens)) ≅ obj
         # Setting what was already there changes nothing.
-@assert set(set(obj, lens, val1), lens, val2) == set(obj, lens, val2)
+@assert set(set(obj, lens, val1), lens, val2) ≅ set(obj, lens, val2)
         # The last set wins.
 
 # output
 
 ```
+Here `≅` is an appropriate notion of equality. In most contexts this is simply `==`. But in some
+contexts it might be `===`, `≈`, `isequal` or something else instead. For instance
+`==` does not work in `Float64` context, because `get(set(obj, lens, NaN), lens) == NaN` can never hold.
+Instead `isequal` or `≅(x::Float64, y::Float64) = isequal(x,y) | x ≈ y` are possible alternatives.
```

Comment:
Maybe explicitly say that "equality **or an approximation of it**" so that it's super clear that the "axioms" of equality/equivalence does not have to hold? (or at least one should not expect that arbitrary lens to satisfy that)

```suggestion
Here `≅` is an appropriate notion of equality or an approximation of it. In most contexts
this is simply `==`. But in some contexts it might be `===`, `≈`, `isequal` or something
else instead. For instance `==` does not work in `Float64` context, because
`get(set(obj, lens, NaN), lens) == NaN` can never hold. Instead `isequal` or
`≅(x::Float64, y::Float64) = isequal(x,y) | x ≈ y` are possible alternatives.
```